### DISCUSSION TITLE
Probe status

### DISCUSSION THREAD

**Discussion by dotnetwizard** [kubernetes-client/csharp#1041] [Q&A]
Is there a way to determine when the main container is ready, like a watch or an event? 

I have 4 containers in a pod, two init containers, one main container, and a finalization container. The main container  takes from 0 to two minutes to get started. I wanted to use a startup probe but wasn't sure of the right way to signal it was ready. What I really want is something like a watch so I can be notified on the server when it is up so I can communicate with it.

**Comment by tg123** [comment]
you could watch the pod, any issue about it?
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes

but I think you may just keep retry connecting to the service without probe it bc retrying is the best probing.

**Reply by dotnetwizard** [reply]
Yes, I have a startup probe with exec cat "some file name" however that is not stopping the server from "doing work on the pod". The server code is just waiting until the container statuses are all running and then pushes work to the pod. In this case the probe could succeed or fail and its status doesn't stop the server. Is there a way to programmatically monitor the probe's status or another indicator that shows that it is not ready?

**Reply by tg123** [reply]
1. put a readyness probe
1. check `pod.spec.container.ready` from api
